subscriber_notifee.go

```diff
@@ -0,0 +1,126 @@
+package dht
+
+import (
+	"github.com/jbenet/goprocess"
+	"github.com/libp2p/go-libp2p-core/event"
+	"github.com/libp2p/go-libp2p-core/network"
+
+	"github.com/libp2p/go-eventbus"
+
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+// subscriberNotifee implements network.Notifee and also manages the subscriber to the event bus. We consume peer
+// identification events to trigger inclusion in the routing table, and we consume Disconnected events to eject peers
+// from it.
+type subscriberNotifee IpfsDHT
+
+func (nn *subscriberNotifee) DHT() *IpfsDHT {
+	return (*IpfsDHT)(nn)
+}
+
+func (nn *subscriberNotifee) subscribe(proc goprocess.Process) {
+	dht := nn.DHT()
+
+	dht.host.Network().Notify(nn)
+	defer dht.host.Network().StopNotify(nn)
```

Comment:
duplicate of https://github.com/libp2p/go-libp2p-kad-dht/pull/472/files#diff-5d01a71abe5aa561a232951a0eb79b0cR125?

Reply:
looks like it, thanks for the catch

Reply:
going to leave this one where it is and remove the one you pointed to.